pkg/volume/glusterfs/glusterfs.go

```diff
@@ -768,11 +766,7 @@ func (p *glusterfsVolumeProvisioner) CreateVolume(gid int) (r *v1.GlusterfsVolum
 		return nil, 0, fmt.Errorf("error [%v] when getting cluster nodes for volume %s", err, volume)
 	}
 
-	// The 'endpointname' is created in form of 'gluster-dynamic-<claimname>'.
-	// createEndpointService() checks for this 'endpoint' existence in PVC's namespace and
-	// If not found, it create an endpoint and svc using the IPs we dynamically picked at time
-	// of volume creation.
-	epServiceName := dynamicEpSvcPrefix + p.options.PVC.Name
+	epServiceName := p.options.PVC.Name
 	epNamespace := p.options.PVC.Namespace
 	endpoint, service, err := p.createEndpointService(epNamespace, epServiceName, dynamicHostIps, p.options.PVC.Name)
```

Comment:
Thanks @dixudx . I think you can replace `epServiceName` directly here with `p.options.PVC.Name`, but we may need to wait some comments from  @msau42 @humblec @saad-ali to see if this can be safely removed.

Reply:
>We can't restrict the PVC name length. This will break backwards compatibility.

> We had a similar name length issue for gce pds, and the workaround that we did was to truncate the generated name. But it is prone to name collisions.

@msau42 So here we adopt the same workaround by truncating `epServiceName`?

Well, I don't think the workaround is a real fix.

Reply:
It's not, but we can't add more restrictions to the API, especially because this affects all PVCs, not just gluster.

Reply:
Completelt agree with @msau42. The plugin should have the provision to choose the name. Changing the name format will actually break backward compatibility, more or less the validation has to be done outside plugin.